src/store/surveys.js

```diff
@@ -127,9 +127,16 @@ export default createReducer(initialState, {
             }
         });
 
-        return state
-            .updateIn(['pendingResponsesByCall', callId, surveyId], survey => survey?
-                survey.mergeDeep(surveyData) : surveyData);
+        const surveyPath = ['pendingResponsesByCall', callId, surveyId];
+        const responsePath = surveyPath.concat(['responses', elemId.toString()])
+        if (state.getIn(responsePath) != null) {
+          return state
+              .updateIn(responsePath, resp => immutable.fromJS(response));
+        } else {
+          return state
+              .updateIn(surveyPath, survey => survey?
+                  survey.mergeDeep(surveyData) : surveyData);
+        }
```

Comment:
Great catch! Am I understanding this code correctly that if there are already responses registered, it will replace them. But if there are no responses, it will merge the entire survey state?

Is there reason to believe that this same issue exists in the activist portal, or is it irrelevant because the framework there is so different?

Reply:
Yes that is how it works.

I tried investigating this but I have a longstanding problem with submitting surveys locally. But I could check on the "live" dev-server and see if the problem exists there. But yes I suspect that it behaves quite differently.